test/integration/user_login_test.rb

```diff
@@ -93,4 +93,76 @@ def setup
     visit admin_orders_path
     assert root_path, current_path
   end
+
+  test "when a user logs in from pending loans, he gets redirected back to
+  pending loans" do
+    user = create(:user)
+    project = create(:project, title: "A Water Purifier")
+
+    visit category_path(project.categories.first.id)
+    within(".row") do
+      click_link_or_button("Lend")
+    end
+    log_in_user(user)
+
+    assert_equal pending_loan_path, current_path
+    assert page.has_content?("Welcome back, #{user.username}")
+  end
+
+  test "when an unauthenticated user selects a loan and logs in from pending
+  loans, he gets redirected back to pending loans with the item in his cart." do
+    user = create(:user)
+    project = create(:project, title: "A Water Purifier")
+
+    visit category_path(project.categories.first.id)
+    within(".row") do
+      click_link_or_button("Lend")
+    end
+    log_in_user(user)
+
+    assert page.has_content?(project.title)
+  end
+
+  test "when an unauthenticated user selects a loan and tries to checkout and
+  logs in from the signup page, he gets redirected back to pending_loan_show and
+  not the signup page" do
+    user = create(:user)
+    project = create(:project, title: "A Water Purifier")
+
+    visit category_path(project.categories.first.id)
+    within(".row") do
+      click_link_or_button("Lend")
+    end
+    click_link_or_button("Checkout")
+    log_in_user(user)
+
+    assert page.has_content?(project.title)
+    refute page.has_content?("Signup Page")
+  end
+
+  test "when a new user selects a loan and tries to checkout and signups, he
+  gets redirected back to the pending_loan_show page" do
+    project = create(:project, title: "A Water Purifier")
+
+    visit category_path(project.categories.first.id)
+    within(".row") do
+      click_link_or_button("Lend")
+    end
+    click_link_or_button("Checkout")
+    fill_in "signup[username]",               with: "Jwan622"
+    fill_in "signup[first_name]",             with: "Jeff"
+    fill_in "signup[last_name]",              with: "Wan"
+    fill_in "signup[street]",                 with: "31 Hillwood Court"
+    fill_in "signup[city]",                   with: "New York City"
+    fill_in "signup[state]",                  with: "NY"
+    fill_in "signup[zipcode]",                with: "10305"
+    fill_in "signup[country]",                with: "USA"
+    fill_in "signup[password]",               with: "password"
+    fill_in "signup[password_confirmation]",  with: "password"
+    fill_in "signup[email]",                  with: "Jwan6221@yahoo.com"
+    click_link_or_button "Create Account"
+    
```

Comment:
Shouldn't we assert that we are redirected to the appropriate path?

assert previous_path, current_path